### PR TITLE
Define regexes outside of functions

### PR DESCRIPTION
**Pull request by crookedneighbor**
Minor optimization. Moves regex creation to variables at the top of the files so new regexs don't need to be created every time the functions run.

### CODE REVIEW

**Comment by crookedneighbor** [comment]
All tests pass.

File: lib/formatter/parse-pattern.js [hunk 1]
```diff
@@ -4,9 +4,10 @@ var ALPHA_REGEX = /[A-Za-z]/;
 var DIGIT_REGEX = /\d/;
 var WILD_REGEX = /./;
 var PLACEHOLDER_REGEX = /^[A-Za-z0-9\*]$/;
-var PLACEHOLDER_PATTERN_REGEX = '({{[^}]+}})';
+var PLACEHOLDER_PATTERN = '({{[^}]+}})';
 var PERMACHAR_REGEX = '(\\s|\\S)';
-var PATTERN_REGEX = new RegExp(PLACEHOLDER_PATTERN_REGEX + '|' + PERMACHAR_REGEX, 'g');
+var PATTERN_REGEX = new RegExp(PLACEHOLDER_PATTERN + '|' + PERMACHAR_REGEX, 'g');
```
Comment (EvanHahn) [inline lib/formatter/parse-pattern.js]: While we're here, we shouldn't be dong `new RegExp`, just `RegExp`.

Comment (crookedneighbor) [inline lib/formatter/parse-pattern.js]: I'll make that change.

**Comment by mrak** [comment]
👍

File: lib/formatter/parse-pattern.js [hunk 2]
```diff
@@ -4,10 +4,11 @@ var ALPHA_REGEX = /[A-Za-z]/;
 var DIGIT_REGEX = /\d/;
 var WILD_REGEX = /./;
 var PLACEHOLDER_REGEX = /^[A-Za-z0-9\*]$/;
-var PLACEHOLDER_PATTERN_REGEX = '({{[^}]+}})';
+var PLACEHOLDER_PATTERN = '({{[^}]+}})';
 var PERMACHAR_REGEX = '(\\s|\\S)';
-var PATTERN_REGEX = new RegExp(PLACEHOLDER_PATTERN_REGEX + '|' + PERMACHAR_REGEX, 'g');
```
Comment (intelliot) [inline lib/formatter/parse-pattern.js]: Was the `new ` here doing anything?

Comment (crookedneighbor) [inline lib/formatter/parse-pattern.js]: It was creating a RegExp object versus creating a RegExp literal. @EvanHahn recommended that I remove the `new`.

**Pull request merged**